SYCL/GroupAlgorithm/SYCL2020/sort.cpp

```diff
@@ -1,369 +1,351 @@
-  // fill data
-  for (std::size_t i = 0; i < dataSize; ++i)
-    stationaryData[i] = CustomType{int(i)};
-
-  // run test
-  for (int test_case = 0; test_case < 1; ++test_case) {
-    for (int sort_case = 0; sort_case < 3; ++sort_case) {
-      run_sort(q, stationaryData, dataSize, CustomFunctor{}, test_case,
-               sort_case);
-    }
+  std::iota(DataReversed.rbegin(), DataReversed.rend(), (size_t)0);
+
+  // Fill using random numbers
+  {
+    std::default_random_engine generator;
+    std::normal_distribution<float> distribution((10.0), (2.0));
+    for (T &Elem : DataRandom)
+      Elem = T(distribution(generator));
   }
-}
 
-template <typename T, typename Compare>
-void test_sort_by_comp(sycl::queue &q, std::size_t dataSize) {
-  std::default_random_engine generator;
-  std::normal_distribution<float> distribution((10.0), (2.0));
-
-  T max_size = std::numeric_limits<T>::max();
-  std::size_t to_fill = dataSize;
-  if (dataSize > max_size)
-    to_fill = max_size;
-
-  // reversed order
-  test_sort_cases<T>()(q, to_fill, Compare{},
-                       [to_fill](std::size_t i) { return T(to_fill - i - 1); });
-  // filled by 1
-  test_sort_cases<T>()(q, dataSize, Compare{},
-                       [](std::size_t) { return T(1); });
-  // random distribution
-  test_sort_cases<T>()(q, dataSize, Compare{},
-                       [&distribution, &generator](std::size_t) {
-                         return T(distribution(generator));
-                       });
-}
+  auto RunOnDataAndComp = [&](const std::vector<T> &Data,
+                              const auto &Comparator) {
+    RunSortOVerGroup<UseGroupT::WorkGroup, 1>(Q, Data, Comparator);
+    RunSortOVerGroup<UseGroupT::WorkGroup, 2>(Q, Data, Comparator);
+
+    RunSortOVerGroup<UseGroupT::SubGroup, 1>(Q, Data, Comparator);
+    RunSortOVerGroup<UseGroupT::SubGroup, 2>(Q, Data, Comparator);
+
+    RunJointSort<UseGroupT::WorkGroup, 1>(Q, Data, Comparator);
+    RunJointSort<UseGroupT::WorkGroup, 2>(Q, Data, Comparator);
 
-template <typename T>
-void test_sort_by_type(sycl::queue &q, std::size_t dataSize) {
-  test_sort_by_comp<T, std::less<T>>(q, dataSize);
-  test_sort_by_comp<T, std::greater<T>>(q, dataSize);
+    RunJointSort<UseGroupT::SubGroup, 1>(Q, Data, Comparator);
+    RunJointSort<UseGroupT::SubGroup, 2>(Q, Data, Comparator);
+  };
+
+  RunOnDataAndComp(DataReversed, std::greater<T>{});
+  RunOnDataAndComp(DataReversed, std::less<T>{});
+  RunOnDataAndComp(DataRandom, std::less<T>{});
+  RunOnDataAndComp(DataRandom, std::greater<T>{});
 }
 
-int main(int argc, char *argv[]) {
-  sycl::queue q(sycl::default_selector_v, async_handler_);
-  if (!isSupportedDevice(q.get_device())) {
-    std::cout << "Skipping test\n";
-    return 0;
-  }
+int main() {
+  try {
+    sycl::queue Q;
+
+    std::vector<size_t> Sizes{18, 64};
+
+    for (size_t Size : Sizes) {
+      RunOverType<std::int32_t>(Q, Size);
```

Comment:
Testing of uint32_t may be not that useful in this test as int32_t is tested already.
IMO, testing of char/short would still be useful to track potential issue for bytes/shorts.

Reply:
Tests for `char` and `short` could be useful as well as all other types, but I'm trying to keep balance between coverage and execution time.

Reply:
Ok, maybe at least keep 'char'? 
The test has 'half' (which is 2-byte type, and kind-of-similar to 'short').

Reply:
Ok, will add `char`.